osu.Game.Tournament/Screens/Editors/LadderEditorScreen.cs

```diff
@@ -30,6 +33,10 @@ public partial class LadderEditorScreen : LadderScreen, IHasContextMenu
 
         private WarningBox rightClickMessage;
 
+        [Resolved]
+        [CanBeNull]
```

Comment:
This - unfortunately - doesn't quite work like you'd like it too. Yes the `[CanBeNull]` technically means that resharper will null-check the property, _however_ `[Resolved]` without `CanBeNull = true` set inside means that DI will crash if it fails to resolve the dialog overlay. So in practice, `dialogOverlay` will either not be null, or the game will crash.

The proper definition is

```csharp
            [Resolved(canBeNull: true)]
            [CanBeNull]
            private IDialogOverlay dialogOverlay { get; set; }
```

With NRT this gets a lot less boilerplatey, but we can't go there yet in tournament client.